backend/corpora/lambdas/api/v1/collection_uuid/upload.py

```diff
@@ -1,21 +1,21 @@
 import requests
 from flask import make_response, g
 
-from .....common.corpora_config import CorporaConfig
-from .....common.corpora_orm import CollectionVisibility, ProcessingStatus
-from .....common import upload_sfn
-from .....common.entities import Collection, Dataset
 from .....api_server.db import dbconnect
+from .....common.upload_sfn import upload
 from .....common.utils.dl_sources.url import MissingHeaderException, from_url
 from .....common.utils.exceptions import (
     ForbiddenHTTPException,
     InvalidParametersHTTPException,
     TooLargeHTTPException,
     MethodNotAllowedException,
     NotFoundHTTPException,
+    MaxFileSizeExceededException,
+    InvalidFileFormatException,
+    NonExistentCollectionException,
+    InvalidProcessingStateException,
+    NonExistentDatasetException,
 )
-from .....common.utils.math_utils import GB
-from backend.corpora.lambdas.api.v1.collection import _owner_or_allowed
 
 
 def link(collection_uuid: str, body: dict, user: str):
```

Comment:
All business logic from this function moved to `backend/corpora/common/upload_sfn.py`'s `upload()` function. This function now only handles translation of `CorporaException`s to HTTP-specific exceptions without changing portal's upload links endpoint behavior.